oddbird/oddvest
# [WIP] Old time entries

## Cute animal pic
_Because everyone likes pictures of animals._

![](https://www.wnpr.org/sites/wnpr/files/styles/x_large/public/201707/fox.jpg)

## Link to Trello card (if applicable)
_If this PR relates to a Trello card, don't forget to reference this PR on the card as well._

https://trello.com/c/xXY9K3D0/22-i-want-better-integration-between-trello-and-harvest

## Description
_The commit messages say what you did; this should explain why and how._

This implements fetching all paginated old time entries and including them in the time report.

Right now it has no caching; I think that should probably be implemented before merging, otherwise this really hammers Harvest's API unnecessarily. (And it's slow, although not so slow as to be unusable.) If you're in a hurry, though, we could merge this as-is and try using it and see if Harvest complains :)

I also did some code cleanup in this diff, replacing `TrelloPromise.all([...]).then(...)` with `await TrelloPromise.all([...])` to flatten out the callbacks and make the code easier to follow and more consistent (`await` everywhere!). As part of this, I got rid of the whole bluebird/TrelloPromise thing; with async we are implicitly using native Promises most places anyway, I don't see any value in jumping through hoops to use bluebird promises in a few places just because Trello does internally.

## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally._

With Oddvest Dev set up, add a task to a card and see the time report is now complete.

---

## Reviewer tasks:
